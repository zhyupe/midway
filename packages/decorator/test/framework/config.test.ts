import {
  Config,
  getClassMetadata,
  INJECT_CUSTOM_TAG,
} from '../../src';

class Test {
  @Config()
  hhh: any;

  @Config('bbb')
  bbb: any;
}

describe('/test/framework/config.test.ts', () => {
  it('config decorator should be ok', () => {
    let data = getClassMetadata(INJECT_CUSTOM_TAG, Test);
    expect(data).toMatchSnapshot();
  });
});
