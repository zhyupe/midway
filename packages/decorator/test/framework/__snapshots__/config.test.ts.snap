// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/framework/config.test.ts config decorator should be ok 1`] = `
Object {
  "bbb": Object {
    "key": "config",
    "propertyName": "bbb",
    "targetKey": "bbb",
  },
  "hhh": Object {
    "key": "config",
    "propertyName": "hhh",
    "targetKey": "hhh",
  },
}
`;
